bundles/org.openhab.core/src/main/java/org/openhab/core/service/WatchQueueReader.java

```diff
@@ -269,75 +315,79 @@ public void run() {
 
                 key.reset();
             } catch (InterruptedException exc) {
-                logger.debug("WatchQueueReader interrupted; thread exiting: {}", exc.getLocalizedMessage());
+                logger.info("Caught InterruptedException. Shutting down.");
                 return;
             } catch (Exception exc) {
-                logger.debug("Exception caught in WatchQueueReader", exc);
+                logger.debug("Exception caught in WatchQueueReader. Restarting.", exc);
             }
         }
     }
 
     /**
-     * Schedules forwarding of the event to the listeners (if appliccable).
-     * <p>
+     * Schedules forwarding of the event to the listeners (if applicable).
+     *
      * By delaying the forwarding, duplicate modification events and those where the actual file-content is not
      * consistent or empty in between will get skipped and the file system gets a chance to "settle" before the
      * framework is going to act on it.
-     * <p>
+     *
      * Also, modification events are received for meta-data changes (e.g. last modification timestamp or file
      * permissions). They are filtered out by comparing the checksums of the file's content.
-     * <p>
-     * See also
-     * <a href=
-     * "https://stackoverflow.com/questions/16777869/java-7-watchservice-ignoring-multiple-occurrences-of-the-same-event">this
-     * discussion</a> on Stack Overflow.
      *
+     * See also <a href=
+     * "https://stackoverflow.com/questions/16777869/java-7-watchservice-ignoring-multiple-occurrences-of-the-same-event">thisdiscussion</a>
+     * on Stack Overflow.
      *
-     * @param key
-     * @param event
-     * @param resolvedPath
-     * @param service
+     * @param key the `WatchKey`
+     * @param event the `WatchEvent` itself
+     * @param resolvedPath the resolved `Path` for this event
+     * @param services a `Set` of `AbstractWatchServices` that subscribe to this event
```

Comment:
JavaDocs are not Markdown, instead of 

```
`WatchKey`
```
 use something like `{@link WatchKey}` or `{@code WatchKey}`